tests/tests/preconfirmations.rs

```diff
@@ -118,6 +119,72 @@ async fn preconfirmation__received_after_successful_execution() {
     ));
 }
 
+#[tokio::test]
+async fn preconfirmation__not_received_when_not_asked() {
```

Comment:
Can we also add `preconfirmation__received_when_asked` please? Just to be sure that this way of testing covers the right behavior=)

Reply:
Done.